x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -542,7 +542,9 @@ function getBaseTemplate({
         _meta,
       },
     },
-    data_stream: { hidden },
+    data_stream: {
+      hidden: registryElasticsearch?.['index_template.data_stream']?.hidden ?? hidden,
```

Comment:
In case `registryElasticsearch?.['index_template.data_stream']?.hidden` is `false` but `hidden` is true, I think it should still result in `data_stream: { hidden: true }`. So if either of the options is `true`, the result is true.

I've added a test case above that I think will fail.

Reply:
Ok, if this is the case then we should use `||` instead of `??`